### PR TITLE
Online users and greetings

### PR DESCRIPTION
**Pull request by PodviaznikovPavlo**
@podviaznikov 

I have finished online users and greetings from bot. You can check it.

### CODE REVIEW

**Comment by PodviaznikovPavlo** [comment]
@podviaznikov 

I have made some questions generation. But there is problems when a more then one tab is open.
For now i havent idea how to load ONE question for all tabs. Now for every new tab bot send new question.

Now, you can check one tab. When your answer is correct bot send another question, if not correct - nothing.

File: app.js [hunk 1]
```diff
@@ -6,11 +6,25 @@ var http = require('http').Server(app);
 var io = require('socket.io')(http);
 
 var questionsList = question.correctQuestions();
+var question;
 
 var users = [];
 var leaders = [{name: 'Anton', points: 50}, {name: 'Pasha', points: 60}, {name: 'Maryna', points: 70}];
 
+function getQuestion(){
+    var generatedQuestion = questionsList[Math.floor(Math.random()*questionsList.length)];
```
Comment (podviaznikov) [inline app.js]: bad var name. This questions is not generated. It's just random.

File: app.js [hunk 2]
```diff
@@ -6,11 +6,25 @@ var http = require('http').Server(app);
...
+    question = generatedQuestion;
+    return generatedQuestion;
+}
 
+function checkTheAnswer(answer){
```
Comment (podviaznikov) [inline app.js:21]: just curious if you know how to write function body in one line?

File: app.js [hunk 3]
```diff
@@ -27,14 +41,24 @@ app.get('/leaders', function(request, response) {
 
 io.on('connection', function(socket){
 
+  io.emit('question', getQuestion());
```
Comment (podviaznikov) [inline app.js]: This is the same problem that was before where I asked you to compare 3 different pieces of code.

Go  back to that question and try to think answer again.

File: app.js [hunk 4]
```diff
@@ -27,14 +41,24 @@ app.get('/leaders', function(request, response) {
...
+    io.emit('online users', users);
+
   socket.on('message', function(msg){
+    var answer = msg.message;
     io.emit('message', msg);
+    io.emit('is answer correct', checkTheAnswer(answer));
+  });
+
+  socket.on('answer is correct', function(){
+    io.emit('question', getQuestion());
```
Comment (podviaznikov) [inline app.js]: same problem as above

File: public/main.js [hunk 5]
```diff
@@ -1,20 +1,37 @@
 $(document).on('ready', function(){
+    var name = '';
 	submitAnswer();
     handleMessage();
     displayLeaders();
+    onlineUsers();
```
Comment (podviaznikov) [inline public/main.js]: inconsistent functions' names. In one case you call `displayLeaders` in another `onlineUsers`. First function name starts with verb, second with noun. In both cases they do very similar thing.

**Comment by podviaznikov** [comment]
```
I have made some questions generation. But there is problems when a more then one tab is open.
For now i havent idea how to load ONE question for all tabs. Now for every new tab bot send new question.
```

@PavelPodviaznikov if you need to send same question to the multiple users means that you need to save that question somewhere. Nothing specific should be done. You are doing things like that already when you send same information for all users.

File: app.js (hunk 1 above)
Comment (PodviaznikovPavlo) [inline app.js]: @podviaznikov 
Okay, ill rename it.

File: app.js (hunk 2 above)
Comment (PodviaznikovPavlo) [inline app.js:21]: @podviaznikov 
I know there is solution. But i forgot how it calls. It has such format a ? b : c
where a - condition, if it true func return b, if false - c.

Did you ask for that thing?

File: app.js (hunk 3 above)
Comment (PodviaznikovPavlo) [inline app.js]: @podviaznikov 
ok

File: public/main.js (hunk 5 above)
Comment (PodviaznikovPavlo) [inline public/main.js]: @podviaznikov 
I'll think about this.

File: app.js (hunk 3 above)
Comment (podviaznikov) [inline app.js]:
```
if(question.answer === answer){
        return true;
    }
    else{
        return false;
    }
```

is the same as
```
return question.answer === answer;
```

File: app.js (hunk 2 above)
Comment (podviaznikov) [inline app.js:21]:
```
if(question.answer === answer){
        return true;
    }
    else{
        return false;
    }
```

is the same as

```
return question.answer === answer;
```

File: app.js [hunk 6]
```diff
@@ -6,11 +6,25 @@ var http = require('http').Server(app);
 var io = require('socket.io')(http);
 
 var questionsList = question.correctQuestions();
```
Comment (podviaznikov) [inline app.js]: why it's called `correctQuestions`? do you have `incorrectQuestions` somewhere too?

Strange name

**Comment by podviaznikov** [comment]
didn't get answers from you.

What is the status of this PR? Should it be merged or not?

**Comment by PodviaznikovPavlo** [comment]
@podviaznikov 

yes, you can merge.

**Pull request merged**